main.py

```diff
@@ -45,6 +45,7 @@ class TestRun:
     file: Path
     result: TestResult
     output: str
+    strict_mode: bool
```

Comment:
```suggestion
    strict_mode: bool | None
```

...as we might return an error result from `run_test()` before having determined `strict_mode` to be `True` or `False`.